examples/runtime_torture/launchable_torture.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2024 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import argparse
+from contextlib import contextmanager
+import sys
+from time import perf_counter
+
+import torch
+import torch.nn as nn
+
+import shark_turbine.aot as aot
+
+from shark_turbine.runtime import (
+    Launchable,
+)
+
+
+@contextmanager
+def report_time(banner):
+    t1 = perf_counter()
+    yield
+    t2 = perf_counter()
+    print(f"{banner}: {(t2 - t1) * 1000}ms")
+
+
+class TestLinearModel(nn.Module):
+    def forward(self, input, weight, bias):
+        result = nn.functional.linear(input, weight, bias)
+        scale = 1 / 256.0
+        result = result * scale
+        return nn.functional.tanh(result)
+
+
+def test_sequential(nreps, device, dtype):
+    print("*** run test_sequential:")
+    print("------------------------")
+    print("*** compiling model")
+    torch.manual_seed(42)
+    initial_input = torch.rand([512, 512], dtype=dtype)
+    weight = torch.rand([512, 512], dtype=dtype)
+    bias = torch.rand([512], dtype=dtype)
+    eo = aot.export(TestLinearModel(), args=(initial_input, weight, bias))
+    launcher = Launchable.jit_compile(str(eo.mlir_module))
+    launcher.preload(device)
+    print("*** preloaded")
+
+    # Compute reference.
+    ref_model = TestLinearModel()
+    ref_input = initial_input.to(device)
+    ref_weight = weight.to(device)
+    ref_bias = bias.to(device)
+    print("*** moved ref inputs to device")
+    with report_time("reference total time"):
+        for i in range(nreps):
```

Comment:
I think we need warm up runs for the reference model? Also later for the turbine model.

Reply:
I wasn't really aiming to benchmark but to bug hunt, so I'll leave this as is.